src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IDynamicInvocationExpression.cs

```diff
@@ -491,5 +491,490 @@ Type Arguments(0)
+          Operand: 
+            IFlowCaptureReferenceOperation: 0 (OperationKind.FlowCaptureReference, Type: C, IsImplicit) (Syntax: 'c1')
+
+    Next (Regular) Block[B2]
+Block[B2] - Block
+    Predecessors: [B1]
+    Statements (1)
+        IFlowCaptureOperation: 1 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'c1')
+          Value: 
+            IFlowCaptureReferenceOperation: 0 (OperationKind.FlowCaptureReference, Type: C, IsImplicit) (Syntax: 'c1')
+
+    Next (Regular) Block[B4]
+Block[B3] - Block
+    Predecessors: [B1]
+    Statements (1)
+        IFlowCaptureOperation: 1 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'c2')
+          Value: 
+            IParameterReferenceOperation: c2 (OperationKind.ParameterReference, Type: C) (Syntax: 'c2')
+
+    Next (Regular) Block[B4]
+Block[B4] - Block
+    Predecessors: [B2] [B3]
+    Statements (1)
+        IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null) (Syntax: '(c1 ?? c2).M2(d);')
+          Expression: 
+            IDynamicInvocationOperation (OperationKind.DynamicInvocation, Type: dynamic) (Syntax: '(c1 ?? c2).M2(d)')
+              Expression: 
+                IDynamicMemberReferenceOperation (Member Name: ""M2"", Containing Type: null) (OperationKind.DynamicMemberReference, Type: null) (Syntax: '(c1 ?? c2).M2')
+                  Type Arguments(0)
+                  Instance Receiver: 
+                    IFlowCaptureReferenceOperation: 1 (OperationKind.FlowCaptureReference, Type: C, IsImplicit) (Syntax: 'c1 ?? c2')
+              Arguments(1):
+                  IParameterReferenceOperation: d (OperationKind.ParameterReference, Type: dynamic) (Syntax: 'd')
+              ArgumentNames(0)
+              ArgumentRefKinds(0)
+
+    Next (Regular) Block[B5]
+Block[B5] - Exit
+    Predecessors: [B4]
+    Statements (0)
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>(source, expectedFlowGraph, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void DynamicInvocation_ControlFlowInFirstArgument()
+        {
+            string source = @"
+class C
+{
+    void M(char ch, C c, dynamic d1, dynamic d2)
+    /*<bind>*/{
+        c.M2(d1 ?? d2, ch);
+    }/*</bind>*/
+
+    public void M2(int i, char ch)
+    {
+    }
+
+    public void M2(long i, char ch)
+    {
+    }
+}
+";
+            string expectedFlowGraph = @"
+Block[B0] - Entry
+    Statements (0)
+    Next (Regular) Block[B1]
+Block[B1] - Block
+    Predecessors: [B0]
+    Statements (2)
+        IFlowCaptureOperation: 0 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'c.M2')
+          Value: 
+            IDynamicMemberReferenceOperation (Member Name: ""M2"", Containing Type: null) (OperationKind.DynamicMemberReference, Type: null) (Syntax: 'c.M2')
```

Comment:
>IDynamicMemberReferenceOperation [](start = 12, length = 32)

Can there be anything else, but an IDynamicMemberReferenceOperation as the IDynamicInvocationOperation.Operation? It would be good to cover scenario like that too.
We should also consider preserving IDynamicMemberReferenceOperation as IDynamicInvocationOperation.Operation instead of spilling it. In that position it is not considered to be a value, please confirm with examining order of evaluation in IL.

Reply:
If we change implementation to handle IDynamicMemberReferenceOperation specially, we'll also need tests with type arguments and containing type.

---
In reply to: [186840169](https://github.com/dotnet/roslyn/pull/26690#discussion_r186840169) [](ancestors = 186840169)

Reply:
Added `DynamicInvocation_ControlFlowWithDynamicTypedMemberReceiver`.

---
In reply to: [186840169](https://github.com/dotnet/roslyn/pull/26690#discussion_r186840169) [](ancestors = 186840169)

Reply:
Verified that we don't need to spill IDynamicMemberReferenceOperation. Updated implementation and test baselines. Also added `DynamicIndexerAccessExpression_ControlFlowWithDynamicReceiver` which verified IDynamicMemberReferenceOperation is not spilled for indexer access.

---
In reply to: [186897472](https://github.com/dotnet/roslyn/pull/26690#discussion_r186897472) [](ancestors = 186897472,186840169)

Reply:
Folded non-zero type arguments and non-null containing type verification in test `DynamicInvocation_ControlFlowNullReceiver`

---
In reply to: [186842102](https://github.com/dotnet/roslyn/pull/26690#discussion_r186842102) [](ancestors = 186842102,186840169)